mac/Keyman4MacIM/Keyman4MacIM/KMCoreActionHandler.m

```diff
@@ -0,0 +1,320 @@
+/**
+ * Keyman is copyright (C) SIL International. MIT License.
+ * 
+ * KMCoreActionHandler.m
+ * Keyman
+ * 
+ * Created by Shawn Schantz on 2023-04-21.
+ * 
+ * Processes an NSArray of CoreAction objects and determines how they should be
+ * applied to the client application. Creates a KMActionHandlerResult object to instruct
+ * Keyman how to apply the changes to the client to conform to the CoreAction
+ * objects.
+ *
+ * To apply the changes in the correct order in the client application, the
+ * pattern in which the actions occur are analyzed and classified with the
+ * ActionPattern enum. Some patterns require a different strategy to preserve
+ * the order.
+ */
+
+#import "KMInputMethodEventHandler.h"
+#import "KMCoreActionHandler.h"
+
+@interface KMCoreActionHandler ()
+
+typedef enum {BackspacesOnly,
+  CharactersOnly,
+  BackspaceBeforeCharacter,
+  CharacterBeforeBackspace,
```

Comment:
I believe I did see a single backspace come from core in my test application in response to typing the backspace key. I can try to reproduce that, but I can't remember what keyboard that was with. I figure we would want that to pass through and mark the event has unhandled if it occurs.

I think all this code goes away as soon as I make the changes to consume the action struct instead of the action list. Or would this still be relevant?

Reply:
Yes, I think the action struct should make all this much simpler